Local testing fails

I'm following the CI setup as described in 

https://github.com/huggingface/datasets/blob/8eee4fa9e133fe873a7993ba746d32ca2b687551/.circleci/config.yml#L16-L19

in a new conda environment, at commit https://github.com/huggingface/datasets/commit/4de6dbf84e93dad97e1000120d6628c88954e5d4

and getting

```
FAILED tests/test_caching.py::RecurseDumpTest::test_dump_ipython_function - TypeError: an integer is required (got type bytes)
1 failed, 2321 passed, 5109 skipped, 10 warnings in 124.32s (0:02:04)
```

Seems like a discrepancy with CI, perhaps a lib version that's not controlled? 
Tried with `pyarrow=={1.0.0,0.17.1,2.0.0}`

--- a/tests/test_caching.py
+++ b/tests/test_caching.py
@@ -1,6 +1,6 @@
 import pickle
 from hashlib import md5
-from types import CodeType, FunctionType
+from types import FunctionType
 from unittest import TestCase
 from unittest.mock import patch
 

--- a/tests/test_caching.py
+++ b/tests/test_caching.py
@@ -145,11 +145,14 @@ class RecurseDumpTest(TestCase):
         )
 
         def create_ipython_func(co_filename, returned_obj):
+            from dill._dill import _create_code
+
             def func():
                 return returned_obj
 
             code = func.__code__
-            code = CodeType(*[getattr(code, k) if k != "co_filename" else co_filename for k in code_args])
+            # Use _create_code from dill in order to make it work for different python versions
+            code = _create_code(*[getattr(code, k) if k != "co_filename" else co_filename for k in code_args])
             return FunctionType(code, func.__globals__, func.__name__, func.__defaults__, func.__closure__)
 
         co_filename, returned_obj = "<ipython-input-2-e0383a102aae>", [0]

--- a/tests/test_caching.py
+++ b/tests/test_caching.py
@@ -1,6 +1,6 @@
 import pickle
 from hashlib import md5
-from types import FunctionType
+from types import CodeType, FunctionType
 from unittest import TestCase
 from unittest.mock import patch
 
@@ -144,9 +144,31 @@ class RecurseDumpTest(TestCase):
             "co_cellvars",
         )
 
-        def create_ipython_func(co_filename, returned_obj):
-            from dill._dill import _create_code
+        def _create_code(*args):
+            """Create CodeType for any python 3 version. From dill._dill._create_code"""
+            if hasattr(args[-3], "encode"):
+                args = list(args)
+                args[-3] = args[-3].encode()  # co_lnotab
+                args[-10] = args[-10].encode()  # co_code
+            if hasattr(CodeType, "co_posonlyargcount"):
+                if len(args) == 16:
+                    return CodeType(*args)
+                elif len(args) == 15:
+                    return CodeType(args[0], 0, *args[1:])
+                return CodeType(args[0], 0, 0, *args[1:])
+            elif hasattr(CodeType, "co_kwonlyargcount"):
+                if len(args) == 16:
+                    return CodeType(args[0], *args[2:])
+                elif len(args) == 15:
+                    return CodeType(*args)
+                return CodeType(args[0], 0, *args[1:])
+            if len(args) == 16:
+                return CodeType(args[0], *args[3:])
+            elif len(args) == 15:
+                return CodeType(args[0], *args[2:])
+            return CodeType(*args)
 
+        def create_ipython_func(co_filename, returned_obj):
             def func():
                 return returned_obj
 

